### PR TITLE
driver.php: updated gcc -v example

### PR DESCRIPTION
**Pull request by denilsonsa**
The original example omitted the important `cc1` step, which was probably an oversight. It also had different temporary filenames between the executions, also an oversight. It also had `==` instead of just `=`, which was a simple typo.

This commit here tries to make the output more faithful (all three steps are visible), while redacting a lot of extra fluff from the output. In fact, we can redact even more:

    gcc -v hello.c 2>&1 | grep '^ [^ ].* '

That will output just the lines starting with a single space, and that contain at least another space somewhere. That's enough to extract the three execution lines from the noisy output. They still have countless parameters in them, which need to be cleaned up manually.

### CODE REVIEW

**Comment by denilsonsa** [comment]
FYI: I've noticed you executed that command on an ARM computer. Well, I tried it on both ARM and x86_64, and the steps are the same (but with different paths and parameters).

**Comment by denilsonsa** [comment]
In fact, upon further inspection, the `clang -v` output is just as noisy as the GCC one. It feels unfair to claim one of them is easier to read than another. Look:

```
clang version 15.0.7
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-pc-linux-gnu/12.2.1
Found candidate GCC installation: /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/12.2.1
Selected GCC installation: /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/12.2.1
Candidate multilib: .;@m64
Candidate multilib: 32;@m32
Selected multilib: .;@m64
 "/usr/bin/clang-15" -cc1 -triple x86_64-pc-linux-gnu -emit-obj -mrelax-all --mrelax-relocations -disable-free -clear-ast-before-backend -disable-llvm-verifier -discard-value-names -main-file-name hello.c -mrelocation-model pic -pic-level 2 -pic-is-pie -mframe-pointer=all -fmath-errno -ffp-contract=on -fno-rounding-math -mconstructor-aliases -funwind-tables=2 -target-cpu x86-64 -tune-cpu generic -mllvm -treat-scalable-fixed-error-as-warning -debugger-tuning=gdb -v -fcoverage-compilation-dir=/home/denilson/Downloads/dc -resource-dir /usr/lib/clang/15.0.7 -internal-isystem /usr/lib/clang/15.0.7/include -internal-isystem /usr/local/include -internal-isystem /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/12.2.1/../../../../x86_64-pc-linux-gnu/include -internal-externc-isystem /include -internal-externc-isystem /usr/include -fdebug-compilation-dir=/home/denilson/Downloads/dc -ferror-limit 19 -stack-protector 2 -fgnuc-version=4.2.1 -faddrsig -D__GCC_HAVE_DWARF2_CFI_ASM=1 -o /tmp/hello-1e59ae.o -x c hello.c
clang -cc1 version 15.0.7 based upon LLVM 15.0.7 default target x86_64-pc-linux-gnu
ignoring nonexistent directory "/usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/12.2.1/../../../../x86_64-pc-linux-gnu/include"
ignoring nonexistent directory "/include"
#include "..." search starts here:
#include <...> search starts here:
 /usr/lib/clang/15.0.7/include
 /usr/local/include
 /usr/include
End of search list.
hello.c:3:1: warning: type specifier missing, defaults to 'int'; ISO C99 and later do not support implicit int [-Wimplicit-int]
main() {
^
int
1 warning generated.
 "/usr/bin/ld" -pie --build-id --eh-frame-hdr -m elf_x86_64 -dynamic-linker /lib64/ld-linux-x86-64.so.2 -o a.out /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/12.2.1/../../../../lib64/Scrt1.o /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/12.2.1/../../../../lib64/crti.o /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/12.2.1/crtbeginS.o -L/usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/12.2.1 -L/usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/12.2.1/../../../../lib64 -L/lib/../lib64 -L/usr/lib/../lib64 -L/usr/bin/../lib -L/lib -L/usr/lib /tmp/hello-1e59ae.o -lgcc --as-needed -lgcc_s --no-as-needed -lc -lgcc --as-needed -lgcc_s --no-as-needed /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/12.2.1/crtendS.o /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/12.2.1/../../../../lib64/crtn.o
```

```
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-pc-linux-gnu/12.2.1/lto-wrapper
Target: x86_64-pc-linux-gnu
Configured with: /build/gcc/src/gcc/configure --enable-languages=c,c++,ada,fortran,go,lto,objc,obj-c++,d --enable-bootstrap --prefix=/usr --libdir=/usr/lib --libexecdir=/usr/lib --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=https://bugs.archlinux.org/ --with-build-config=bootstrap-lto --with-linker-hash-style=gnu --with-system-zlib --enable-__cxa_atexit --enable-cet=auto --enable-checking=release --enable-clocale=gnu --enable-default-pie --enable-default-ssp --enable-gnu-indirect-function --enable-gnu-unique-object --enable-libstdcxx-backtrace --enable-link-serialization=1 --enable-linker-build-id --enable-lto --enable-multilib --enable-plugin --enable-shared --enable-threads=posix --disable-libssp --disable-libstdcxx-pch --disable-werror
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 12.2.1 20230201 (GCC) 
COLLECT_GCC_OPTIONS='-v' '-mtune=generic' '-march=x86-64' '-dumpdir' 'a-'
 /usr/lib/gcc/x86_64-pc-linux-gnu/12.2.1/cc1 -quiet -v hello.c -quiet -dumpdir a- -dumpbase hello.c -dumpbase-ext .c -mtune=generic -march=x86-64 -version -o /tmp/ccuMa8Pj.s
GNU C17 (GCC) version 12.2.1 20230201 (x86_64-pc-linux-gnu)
	compiled by GNU C version 12.2.1 20230201, GMP version 6.2.1, MPFR version 4.2.0, MPC version 1.3.1, isl version isl-0.26-GMP

GGC heuristics: --param ggc-min-expand=100 --param ggc-min-heapsize=131072
ignoring nonexistent directory "/usr/lib/gcc/x86_64-pc-linux-gnu/12.2.1/../../../../x86_64-pc-linux-gnu/include"
#include "..." search starts here:
#include <...> search starts here:
 /usr/lib/gcc/x86_64-pc-linux-gnu/12.2.1/include
 /usr/local/include
 /usr/lib/gcc/x86_64-pc-linux-gnu/12.2.1/include-fixed
 /usr/include
End of search list.
GNU C17 (GCC) version 12.2.1 20230201 (x86_64-pc-linux-gnu)
	compiled by GNU C version 12.2.1 20230201, GMP version 6.2.1, MPFR version 4.2.0, MPC version 1.3.1, isl version isl-0.26-GMP

GGC heuristics: --param ggc-min-expand=100 --param ggc-min-heapsize=131072
Compiler executable checksum: 99df1fbb4c4d7eeb8d93f0979bfab9ce
hello.c:3:1: warning: return type defaults to ‘int’ [-Wimplicit-int]
    3 | main() {
      | ^~~~
COLLECT_GCC_OPTIONS='-v' '-mtune=generic' '-march=x86-64' '-dumpdir' 'a-'
 as -v --64 -o /tmp/ccYNnNNi.o /tmp/ccuMa8Pj.s
GNU assembler version 2.40 (x86_64-pc-linux-gnu) using BFD version (GNU Binutils) 2.40
COMPILER_PATH=/usr/lib/gcc/x86_64-pc-linux-gnu/12.2.1/:/usr/lib/gcc/x86_64-pc-linux-gnu/12.2.1/:/usr/lib/gcc/x86_64-pc-linux-gnu/:/usr/lib/gcc/x86_64-pc-linux-gnu/12.2.1/:/usr/lib/gcc/x86_64-pc-linux-gnu/
LIBRARY_PATH=/usr/lib/gcc/x86_64-pc-linux-gnu/12.2.1/:/usr/lib/gcc/x86_64-pc-linux-gnu/12.2.1/../../../../lib/:/lib/../lib/:/usr/lib/../lib/:/usr/lib/gcc/x86_64-pc-linux-gnu/12.2.1/../../../:/lib/:/usr/lib/
COLLECT_GCC_OPTIONS='-v' '-mtune=generic' '-march=x86-64' '-dumpdir' 'a.'
 /usr/lib/gcc/x86_64-pc-linux-gnu/12.2.1/collect2 -plugin /usr/lib/gcc/x86_64-pc-linux-gnu/12.2.1/liblto_plugin.so -plugin-opt=/usr/lib/gcc/x86_64-pc-linux-gnu/12.2.1/lto-wrapper -plugin-opt=-fresolution=/tmp/cc2KNoNJ.res -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s -plugin-opt=-pass-through=-lc -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s --build-id --eh-frame-hdr --hash-style=gnu -m elf_x86_64 -dynamic-linker /lib64/ld-linux-x86-64.so.2 -pie /usr/lib/gcc/x86_64-pc-linux-gnu/12.2.1/../../../../lib/Scrt1.o /usr/lib/gcc/x86_64-pc-linux-gnu/12.2.1/../../../../lib/crti.o /usr/lib/gcc/x86_64-pc-linux-gnu/12.2.1/crtbeginS.o -L/usr/lib/gcc/x86_64-pc-linux-gnu/12.2.1 -L/usr/lib/gcc/x86_64-pc-linux-gnu/12.2.1/../../../../lib -L/lib/../lib -L/usr/lib/../lib -L/usr/lib/gcc/x86_64-pc-linux-gnu/12.2.1/../../.. /tmp/ccYNnNNi.o -lgcc --push-state --as-needed -lgcc_s --pop-state -lc -lgcc --push-state --as-needed -lgcc_s --pop-state /usr/lib/gcc/x86_64-pc-linux-gnu/12.2.1/crtendS.o /usr/lib/gcc/x86_64-pc-linux-gnu/12.2.1/../../../../lib/crtn.o
COLLECT_GCC_OPTIONS='-v' '-mtune=generic' '-march=x86-64' '-dumpdir' 'a.'
```

If the clang output is being cleaned up for the educational purposes of the article, I think the same should happen to the gcc output.

File: driver.php [hunk 1]
```diff
@@ -203,14 +203,17 @@
...
+COLLECT_GCC_OPTIONS=...
+ /usr/lib/gcc/aarch64-linux-gnu/11/cc1 hello.c ... -o /tmp/ccCKUy4c.s
 COLLECT_GCC_OPTIONS=...
  as -v -EL -mabi=lp64 -o /tmp/ccUkKMFH.o /tmp/ccCKUy4c.s
-COMPILER_PATH==...
+COMPILER_PATH=...
+LIBRARY_PATH=...
 COLLECT_GCC_OPTIONS=...
- /usr/lib/gcc/aarch64-linux-gnu/11/collect2 /tmp/ccIPliNQ.o
+ /usr/lib/gcc/aarch64-linux-gnu/11/collect2 ... /tmp/ccUkKMFH.o ...
```
Comment (fabiensanglard) [inline driver.php]: The intro explains that outputs are modified for typesetting purposes. I assume you wanted to add this ellipsis to explain that there is more stuff here but I find it more visually confusing.

File: driver.php [hunk 2]
```diff
@@ -203,14 +203,17 @@
...
 </pre>
 <p>
-It is a lot harder to read but we can still make out that an assembler step <code>as</code> took place, followed by a linking step via <code>collect2</code>. The call to the compiler via flag <code>-cc1</code> however is not featured.
+It is a lot harder to read but we can still make out that a compiler step <code>cc1</code> took place, followed by an assembler step <code>as</code>, followed by a linking step via <code>collect2</code>.
```
Comment (fabiensanglard) [inline driver.php]: This is a good. If you send a separate PR, I will merge it (I disagree with the change above).

File: driver.php [hunk 3]
```diff
@@ -203,17 +203,12 @@
...
-LIBRARY_PATH=...
-COLLECT_GCC_OPTIONS=...
- /usr/lib/gcc/aarch64-linux-gnu/11/collect2 ... /tmp/ccUkKMFH.o ...
+ /usr/lib/gcc/aarch64-linux-gnu/11/cc1 hello.c -o /tmp/ccCKUy4c.s
+ as -o /tmp/ccUkKMFH.o /tmp/ccCKUy4c.s
+ /usr/lib/gcc/aarch64-linux-gnu/11/collect2 /tmp/ccUkKMFH.o
 </pre>
 <p>
-It is a lot harder to read but we can still make out that a compiler step <code>cc1</code> took place, followed by an assembler step <code>as</code>, followed by a linking step via <code>collect2</code>.
+We can still make out that a compiler step <code>cc1</code> took place, followed by an assembler step <code>as</code>, followed by a linking step via <code>collect2</code>.
```
Comment (fabiensanglard) [inline driver.php:211]: I maintain that `gcc` output  is harder to make sense of. You see a "collect" step and it is not obvious what is going on there.

File: driver.php (hunk 1 above)
Comment (denilsonsa) [reply driver.php]: Agreed.

**Comment by denilsonsa** [comment]
After @fabiensanglard 's comments, I'm closing this PR here in favor of #25.

**Pull request closed without merging**